### ISSUE TITLE
User guide

### ISSUE THREAD

**Issue by lebigot** [prabirshrestha/vim-lsp#591]
As a user (not versed in vimscript), I can't figure out how to use this module: may I suggest that it could be used by many more people if there was a user guide, that starts with the most commonly used features?

For example, I had no idea of how to do completion in vim-lsp: the provided links are not clear.

Another example is warnings: they are active by default, but how to learn more about each warning and navigate them?

Also, it would be nice to learn upfront that vim-lsp does intelligent completion because the code is analyzed (with class inference)—maybe it's somewhere in the LSP documentation, but who knows if vim-lsp handles this? Another example is: I have this project with multiple files: will the completion work across Python files?

Etc., etc.

The current README dives immediately in less important technical details, before explaining to the majority of users what vim-lsp is good for.

In a nutshell, starting the README by advertising what asp-vim brings in terms of very important, everyday functionalities, and explaining it in the most simple terms could draw many more users in (you could count me in, but for the moment I'm not sure at all I am ready to invest hours just to learn how to do basic things :) ).
